forge-game/src/main/java/forge/game/zone/MagicStack.java

```diff
@@ -637,71 +637,62 @@ public final boolean hasLegalTargeting(final SpellAbility sa) {
-                        invalidTarget = si == null ? true : !sa.canTarget(si.getSpellAbility());
-                    } else {
-                        invalidTarget = !sa.canTarget(o);
-                    }
-                    // TODO remove targets only after the Loop
-                    // Remove targets
-                    if (invalidTarget) {
-                        choices.remove(o);
-                    } else {
-                        fizzle = false;
+        List<GameObject> toRemove = Lists.newArrayList();
+        if (sa.usesTargeting() && !sa.isZeroTargets()) {
+            fizzle = true;
+            // Some targets were chosen, fizzling for this subability is now possible
+            //fizzle = true;
+            // With multi-targets, as long as one target is still legal,
+            // we'll try to go through as much as possible
+            for (final GameObject o : sa.getTargets()) {
+                boolean invalidTarget = false;
+                if (rememberTgt) {
+                    source.addRemembered(o);
+                }
+                if (o instanceof Card) {
+                    final Card card = (Card) o;
+                    Card current = game.getCardState(card);
+                    if (current != null) {
+                        invalidTarget = !current.equalsWithGameTimestamp(card);
                     }
+                    invalidTarget = invalidTarget || !sa.canTarget(card);
+                } else if (o instanceof SpellAbility) {
+                    SpellAbilityStackInstance si = getInstanceMatchingSpellAbilityID((SpellAbility)o);
+                    invalidTarget = si == null ? true : !sa.canTarget(si.getSpellAbility());
+                } else {
+                    invalidTarget = !sa.canTarget(o);
+                }
 
-                    if (sa.hasParam("CantFizzle")) {
-                        // Gilded Drake cannot be countered by rules if the
-                        // targeted card is not valid
-                        fizzle = false;
-                    }
+                if (invalidTarget) {
+                    toRemove.add(o);
+                } else {
+                    fizzle = false;
                 }
-                if (fizzle == null) {
-                    fizzle = true;
+
+                if (sa.hasParam("CantFizzle")) {
+                    // Gilded Drake cannot be countered by rules if the
+                    // targeted card is not valid
+                    fizzle = false;
                 }
             }
         }
-        else if (sa.getTargetCard() != null) {
-            fizzle = !sa.canTarget(sa.getTargetCard());
-        } else {
-            // Set fizzle to the same as the parent if there's no target info
-            fizzle = parentFizzled;
+        if (sa.getSubAbility() != null) {
+            fizzle = hasFizzled(sa.getSubAbility(), source) && fizzle;
```

Comment:
your refactor is wrong, I don't like that no test catch that
e. g. _Annihilate_ still resolves when target gains hexproof

Reply:
Ugh can you try to update it?

And if possible add a test case for this?

Reply:
- [ ] I want to check if there are some cards where it should still fizzle from ParentTarget (maybe because they got scripted weirdly)

Reply:
I found _Breaking of the Fellowship_ which also fizzles on master if only the first target becomes illegal 💩

Reply:
Fixed now